File: test-suite-kotlin2-ksp2/src/test/kotlin/io/micronaut/docs/server/response/StatusControllerSpec.kt
package io.micronaut.docs.server.response

import io.kotest.matchers.shouldBe
import io.kotest.core.spec.style.StringSpec
import io.micronaut.context.ApplicationContext
import io.micronaut.http.HttpRequest
import io.micronaut.http.HttpStatus
import io.micronaut.http.client.HttpClient
import io.micronaut.runtime.server.EmbeddedServer

class StatusControllerSpec: StringSpec() {

    val embeddedServer = autoClose(
            ApplicationContext.run(EmbeddedServer::class.java, mapOf("spec.name" to "httpstatus"))
    )

    val client = autoClose(
            embeddedServer.applicationContext.createBean(HttpClient::class.java, embeddedServer.getURL())
    )

    init {
        "test status"() {
            var response = client.toBlocking().exchange(HttpRequest.GET<Any>("/status"), String::class.java)
            var body = response.body

            response.status shouldBe HttpStatus.CREATED
            body.get() shouldBe "success"

            response = client.toBlocking().exchange(HttpRequest.GET<Any>("/status/http-response"), String::class.java)
            body = response.body

            response.status shouldBe HttpStatus.CREATED
            body.get() shouldBe "success"

            response = client.toBlocking().exchange(HttpRequest.GET<Any>("/status/http-status"), String::class.java)

            response.status shouldBe HttpStatus.CREATED
        }
    }

}

File: test-suite-python/src/test/python/micronaut/docs/server/response/StatusControllerSpec.py
from typing import Annotated

import java
from jakarta.inject import Inject
from micronaut.context.annotation import Property
from micronaut.http.client import HttpClient
from micronaut.http.client.annotation import Client
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test

HttpRequest = java.type("io.micronaut.http.HttpRequest")
HttpStatus = java.type("io.micronaut.http.HttpStatus")
String = java.type("java.lang.String")


@Property(name="spec.name", value="httpstatus")
@MicronautTest
class StatusControllerSpec:
    client: Annotated[HttpClient, Inject, Client("/")]

    @Test
    def test_status(self):
        response = self.client.toBlocking().exchange(HttpRequest.GET("/status"), String)
        body = response.getBody()

        assert response.getStatus() == HttpStatus.CREATED
        assert body.get() == "success"

        response = self.client.toBlocking().exchange(HttpRequest.GET("/status/http-response"), String)
        body = response.getBody()

        assert response.getStatus() == HttpStatus.CREATED
        assert body.get() == "success"

        response = self.client.toBlocking().exchange(HttpRequest.GET("/status/http-status"), String)

        assert response.getStatus() == HttpStatus.CREATED
